src/lib/construct/graphql-api-lambda-with-cache/main.ts

```diff
@@ -0,0 +1,133 @@
+import * as cdk from 'aws-cdk-lib'
+import * as ec2 from 'aws-cdk-lib/aws-ec2'
+import * as elasticache from 'aws-cdk-lib/aws-elasticache'
+import * as iam from 'aws-cdk-lib/aws-iam'
+import { Construct } from 'constructs'
+import { GraphQLApiLambda } from '..'
+import { GraphQlApiLambdaWithCacheProps } from '../../types'
+
+/**
+ * @stability stable
+ * @category cdk-utils.graphql-api-lambda-with-cache
+ * @subcategory construct
+ * @classdesc Provides a construct to create and deploy a Graphql API as Lambda with Caching
+ *
+ * <b>Architecture</b> ![Architecture](./GraphQLApiLambda.jpg)
+ *
+ * @example
+ * import { GraphQlApiLambdaWithCacheProps } '@gradientedge/cdk-utils'
+ * import { Construct } from 'constructs'
+ *
+ * class CustomConstruct extends GraphQLApiLambdaWithCache {
+ *   constructor(parent: Construct, id: string, props: GraphQlApiLambdaWithCacheProps) {
+ *     super(parent, id, props)
+ *     this.props = props
+ *     this.id = id
+ *     this.initResources()
+ *   }
+ * }
+ */
+export class GraphQLApiLambdaWithCache extends GraphQLApiLambda {
+  /* graphql restApi props */
+  props: GraphQlApiLambdaWithCacheProps
+  id: string
+
+  /* graphql restApi resources */
+  graphQLVpc: ec2.Vpc
+  graphQLElastiCache: elasticache.CfnCacheCluster
+  graphQLSecurityGroup: ec2.ISecurityGroup
+  securityGroupStackName: string
+  securityGroupExportName: string
+
+  constructor(parent: Construct, id: string, props: GraphQlApiLambdaWithCacheProps) {
+    super(parent, id, props)
+
+    this.props = props
+    this.id = id
+  }
+
+  protected initResources() {
+    this.setVpc()
+    this.setSecurityGroup()
+    this.createElastiCache()
+    super.initResources()
+  }
+
+  /**
+   * Create VPC
+   * @protected
+   */
+  protected setVpc() {
+    this.graphQLVpc = this.vpcManager.createCommonVpc(this, this.props.graphQLVpc, this.props.graphQLVpc.vpcName)
+  }
+
+  /**
+   * Resolve Security Group
+   * @protected
+   */
+  protected setSecurityGroup() {
+    this.graphQLSecurityGroup = ec2.SecurityGroup.fromSecurityGroupId(
+      this,
+      `${this.id}`,
+      cdk.Fn.importValue(`${this.securityGroupStackName}-${this.props.stage}-${this.securityGroupExportName}`)
```

Comment:
this construct should be generic and not depend on there stacks. 
... The `securityGroupExportName` should be passed as props instead of member var
can be simplified to just `cdk.Fn.importValue(this.props.securityGroupExportName)` where you set the `securityGroupExportName` in the inheriting construct with the stack, stage et. al